libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Skills/SkillHttpClient.cs

```diff
@@ -25,8 +25,39 @@ public SkillHttpClient(HttpClient httpClient, ICredentialProvider credentialProv
             _conversationIdFactory = conversationIdFactory;
         }
 
+        /// <summary>
+        /// Uses the SkillConversationIdFactory to create or retrieve a Skill Conversation Id, and sends the activity.
+        /// </summary>
+        /// <param name="originatingAudience">The oauth audience scope, used during token retrieval. (Either https://api.botframework.com or bot app id.)</param>
+        /// <param name="fromBotId">The MicrosoftAppId of the bot sending the activity.</param>
+        /// <param name="toSkill">The skill to create the conversation Id for.</param>
+        /// <param name="callbackUrl">The callback Url for the skill host.</param>
+        /// <param name="activity">The activity to send.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        /// <returns>Async task with invokeResponse.</returns>
+        public async Task<InvokeResponse> PostActivityAsync(string originatingAudience, string fromBotId, BotFrameworkSkill toSkill, Uri callbackUrl, Activity activity, CancellationToken cancellationToken)
+        {
+            string skillConversationId;
+            if (_conversationIdFactory is SkillConversationIdFactoryExBase idFactory)
+            {
+                skillConversationId = await idFactory.CreateSkillConversationIdAsync(originatingAudience, fromBotId, activity, toSkill, CancellationToken.None).ConfigureAwait(false);
```

Comment:
This (as I'm sure you know) isn't great, because if we require the `originatingAudience` but the factory is the old abstract class, then the developer has to pass in something that will not be needed.

Which is an distantly-related argument for passing in a `ClaimsIdentity` instead of the `originatingAudience` and `fromBotId`.